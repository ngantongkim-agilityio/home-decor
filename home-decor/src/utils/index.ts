export * from './products';
export * from './notifications';
