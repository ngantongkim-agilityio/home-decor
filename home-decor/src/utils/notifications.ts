import * as Notifications from 'expo-notifications';

export const requestPermissionsNotifications = async () => {
  await Notifications.requestPermissionsAsync();
  Notifications.setNotificationHandler({
    handleNotification: async () => ({
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: false,
    }),
  });
};

export const scheduleNotification = async ({
  id = '',
  title = `You've got a new notification! ðŸ“¬`,
  body = 'Here is the notification message',
  seconds = 5,
}) =>
  await Notifications.scheduleNotificationAsync({
    content: { title, body, data: { url: `/details/${id}` } },
    trigger: {
      seconds,
      repeats: false,
    } as Notifications.TimeIntervalTriggerInput,
  });
