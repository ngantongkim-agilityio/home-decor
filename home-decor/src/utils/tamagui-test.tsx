import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { TamaguiProvider } from 'tamagui';
import { tamaguiConfig } from '@/themes/tamagui.config';

const Providers = ({ children }: PropsWithChildren) => {
  return <TamaguiProvider config={tamaguiConfig}>{children}</TamaguiProvider>;
};

const baseRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';
export { baseRender as render };
