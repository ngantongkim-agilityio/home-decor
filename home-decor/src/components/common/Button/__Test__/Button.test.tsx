import { fireEvent, render } from '@/utils/tamagui-test';

import { Button } from '..';

describe('Button Component', () => {
  it('Should calls onPress when pressed', () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <Button onPress={mockOnPress} disabled>
        Test Button
      </Button>,
    );
    const button = getByText('Test Button');
    fireEvent.press(button);

    expect(mockOnPress).toHaveBeenCalled();
  });

  // it('should not call onPress function while disabled', () => {
  //   const mockOnPress = jest.fn();
  //   const { getByTestId } = render(
  //     <Button onPress={mockOnPress} disabled isLoading>
  //       Test Button
  //     </Button>,
  //   );

  //   const button = getByTestId('button');
  //   console.log(button);
  //   fireEvent.press(button);

  //   expect(mockOnPress).toHaveBeenCalledTimes(0);
  // });

  // it('should not call function onPress while loading', () => {
  //   const mockOnPress = jest.fn();
  //   const { getByTestId } = render(
  //     <Button onPress={handlePress} isLoading>
  //       Test Button
  //     </Button>,
  //   );

  //   const button = getByTestId('button');
  //   fireEvent.press(button);

  //   expect(mockOnPress).toHaveBeenCalledTimes(0);
  // });
});
