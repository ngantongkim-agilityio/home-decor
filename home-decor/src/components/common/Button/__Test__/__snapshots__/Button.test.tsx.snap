// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render to match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f4b5a4",
      "borderBottomLeftRadius": 50,
      "borderBottomRightRadius": 50,
      "borderBottomWidth": 0,
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderStyle": "solid",
      "borderTopLeftRadius": 50,
      "borderTopRightRadius": 50,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "gap": 10,
      "height": 41,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#cc7862",
          "fontFamily": "PoppinsBold",
          "fontSize": 15,
          "lineHeight": 21,
          "textDecorationLine": "none",
        }
      }
      suppressHighlighting={true}
    />
  </View>
</View>
`;
